src/sagemaker/workflow/steps.py

```diff
@@ -93,6 +94,26 @@ def ref(self) -> Dict[str, str]:
         return {"Name": self.name}
 
 
+@attr.s
+class CacheConfig:
+    """Step to cache pipeline workflow.
+
+    Attributes:
+        enable_caching (bool): To enable step caching. Off by default.
+        expire_after (str): If step caching is enabled, a timeout also needs to defined.
+            It defines how old a previous execution can be to be considered for reuse.
+            Needs to be ISO 8601 duration string.
+    """
+
+    enable_caching: bool = attr.ib(default=False)
+    expire_after: str = attr.ib(factory=str)
```

Comment:
Do we need to add logic to have this fail fast if the user enables caching but doesn't provide an input?

Reply:
pipeline service team checks for the invalid inputs? Should we include that too?